### PR TITLE
Enable uri schemes

### PR DESCRIPTION
**Pull request by rwols**
This attaches jdtls to views with a `jdt:` URI scheme. Depends on https://github.com/sublimelsp/LSP/pull/1758

### CODE REVIEW

**Comment by rwols** [comment]
... and 1ccb871 defines on_open_uri_async to handle `jdt:` with the custom `java/classFileContents` request.

**Comment by rwols** [comment]
Not sure if it actually works. Just went ahead with this given the code in LSP-Deno. Someone with access to a debugger would have to verify this working.

**Review by rchl — APPROVED** [review]

**Comment by LDAP** [comment]
Something does not work as expected:
grafik

This seems like a Debugger issue (maybe @daveleroy can help with this).

**Comment by rwols** [comment]
I guess if you only have an URI in the debugger, you can't actually open it via the session.open_location_async machinery. We would need a window command for that.

File: plugin.py [hunk 1]
```diff
@@ -156,6 +159,20 @@ def install_or_update(cls) -> None:
...
 
+    def on_open_uri_async(self, uri: DocumentUri, callback: Callable[[str, str, str], None]) -> bool:
+        if not uri.startswith("jdt:"):
+            return False
+        session = self.weaksession()
+        if not session:
+            return False
+        view = session.window.active_view()
+        session.send_request_async(
+            Request("java/classFileContents", text_document_identifier(uri), view, progress=True),
```
[the remark this answers is not in the document]
Comment (rwols) [reply plugin.py]: made it into a window request

**Comment by rwols** [comment]
Given a `path` in the `stackTrace` response:

```
'path': 'jdt://contents/java.base/java.lang/Thread.class?=jdt.ls-java-project/D:%5C/Local_Repositories%5C/windows-dev-tools%5C/bins%5C/jdk%5C/lib%5C/jrt-fs.jar%60java.base=/javadoc_location=/https:%5C/%5C/docs.oracle.com%5C/en%5C/java%5C/javase%5C/15%5C/docs%5C/api%5C/=/%3Cjava.lang(Thread.class'
```

you then need to invoke the ST view command

```
lsp_open_location
```

with arguments

```js
{
    "location":
    {
        "uri": "jdt://contents/...",
        "range":
        {
            "start":
            {
                "line": 841,  // not sure if DAP is 1-based or 0-based; LSP is 0-based.
                "character": 1 // not sure if DAP is 1-based or 0-based; LSP is 0-based.
            },
            "end":
            {
                "line": 841, // not sure if DAP is 1-based or 0-based; LSP is 0-based.
                "character": 1 // not sure if DAP is 1-based or 0-based; LSP is 0-based.
            }
        }
    },
    "session_name": "jdtls"
}
```

This invocation should be done from within the DAP package.

**Pull request merged**